core/src/test/java/org/apache/iceberg/TestManifestCaching.java

```diff
@@ -178,27 +182,27 @@ public void testWeakFileIOReferenceCleanUp() {
 
     System.gc();
     manifestCache.cleanUp();
-    awaitQuiescence();
-    Assert.assertEquals(maxIO, manifestCache.estimatedSize());
     Assert.assertEquals(maxIO, manifestCache.stats().loadCount());
-    Assert.assertEquals(
-        "No entries should be evicted before IO_MANIFEST_CACHE_MAX_FILEIO_DEFAULT exceeded.",
-        0,
-        manifestCache.stats().evictionCount());
 
     // Insert one more FileIO to trigger cache eviction.
     FileIO lastIO = cacheEnabledHadoopFileIO();
     ContentCache lastCache = contentCache(manifestCache, lastIO);
-    System.gc();
-    manifestCache.cleanUp();
-    awaitQuiescence();
+
+    int numGC = 0;
+    int maxGC = 100;
+    while (manifestCache.estimatedSize() > 2 && numGC < maxGC) {
+      System.gc();
```

Comment:
in general it's not a good practice to depend on repeated (or even single) `System.gc()` calls to make something work.
I would probably suggest to use https://github.com/google/guava/blob/master/guava-testlib/src/com/google/common/testing/GcFinalization.java#L297 once here instead of plain `System.gc()` calls in a loop